### DISCUSSION TITLE
Auto shutdown after print finishes

### DISCUSSION THREAD

**Discussion by taker218** [OctoFarm/OctoFarm#1341] [Feature Request]
I'm currently using the octoprint plugin [Shutdown Printer](https://github.com/devildant/OctoPrint-ShutdownPrinter) to automatically shutdown my printer, after the print is done.

Sometimes I don't want that to happen and for that the plugin has a checkbox you can tick/untick
I do almost everything in OctoFarm now, but I need to go to the OctoPrint webinterface everytime I need to change this setting.

I don't know if the plugin supports changing the setting via api (or if it is even maintained anymore, since the latest release is from 2019), but maybe it would be possible to integrate something like this into OctoFarm itself?

**Comment by NotExpectedYet** [comment]
> I'm currently using the octoprint plugin [Shutdown Printer](https://github.com/devildant/OctoPrint-ShutdownPrinter) to automatically shutdown my printer, after the print is done.
> 
> Sometimes I don't want that to happen and for that the plugin has a checkbox you can tick/untick I do almost everything in OctoFarm now, but I need to go to the OctoPrint webinterface everytime I need to change this setting.
> 
> I don't know if the plugin supports changing the setting via api (or if it is even maintained anymore, since the latest release is from 2019), but maybe it would be possible to integrate something like this into OctoFarm itself?

There will be a call made by the UI when you press and save that button I reckon. Majority of the plugins use the AI to communicate to the OctoPrint backend. 

It's certainly possible, I'm just warey of adding support for another plugin and this is pretty specific to that one.

Happy for idea's here, just thinking out loud but replicating / adding to the power functionality with an extra endpoint to call might be better / quicker to implement now. If that's filled in/enabled there could be two print buttons. Print and Shutdown, or just Print.

**Reply by taker218** [reply]
yeah, I just checked.

There is a call to /api/plugin/shutdownprinter with a Post:
{"command":"disable","eventView":false} or {"command":"enable","eventView":false}

But I think making this available in OctoFarm itself would be a lot nicer, since you wouldn't need to rely on another plugin.

If you are going to implement this into OctoFarm itself, it would be nice to have a target temperature setting. In the plugin the shutdown get's triggered, once the hotend cooled down to a specific temperature the user inputs into the settings.

**Reply by NotExpectedYet** [reply]
That is a good thing to consider, as it would need to look for the temperature to trigger the off safely. Thankfully I can use this:
![image](https://user-images.githubusercontent.com/61137988/206185229-60ef12cf-eec9-494b-9ab9-92f740186cdc.png)

That option is currently used to trigger off the scripts feature so this shouldn't be too hard to add in. 

So rather than dual buttons I'm thinking to do a little dropdown on the end of the start button. Default will always be "Start Print", changes could be rememebered on the browser so new top action becomes whatever the last action was. This could be enabled in that list if that makes sense. That would also set me up for another feature I've been planning to allow for maintenance prints/test prints from the UI to pre-mark them in History.
